src/main/java/chess/service/ChessService.java

```diff
@@ -26,112 +38,109 @@ public ChessService(BoardDao boardDao, RoomDao roomDao) {
         this.roomDao = roomDao;
     }
 
-    public BoardDto getInitialBoard(long roomId) {
-        GameState gameState = getGameState(roomId);
-        Map<String, String> board = getBoardPieces(roomId);
-        List<PieceDto> pieces = board.entrySet()
-                .stream()
-                .map(i -> new PieceDto(i.getKey(), i.getValue()))
-                .collect(Collectors.toUnmodifiableList());
-        return new BoardDto(pieces, gameState.getTeam());
+    public void createRoom(CreateRoomDto room) {
+        Long roomId = roomDao.save(room.getTitle(), room.getPassword());
+        boardDao.saveAll(BoardInitialize.create(), roomId);
     }
 
-    private GameState getGameState(long roomId) {
-        RoomDto room = roomDao.findById(roomId);
-        if (room == null) {
-            return createGameState(roomId);
+    public List<RoomDto> findRoomList() {
+        List<Room> rooms = roomDao.findAll();
+        return rooms.stream()
+            .map(room -> new RoomDto(room.getId(), room.getTeam(), room.getTitle(),
+                room.getPassword(), room.getStatus()))
+            .collect(Collectors.toList());
+    }
+
+    public void deleteBy(Long roomId, String password) {
```

Comment:
1. 유효성 검사를 먼저 진행하고 게임 로직을 처리하도록 만들어보면 어떨까요?
2. room에서 비밀번호 값을 꺼내서 검사하지 말고, 객체에게 비밀번호가 맞는지 물어보도록 수정해보아요. 메서드명은 isSame 대신 적절한 이름으로 바꿔보아요

```suggestion
    public void deleteBy(Long roomId, String password) {
        Room room = roomDao.findById(roomId);
        if (room.getStatus()) {
            throw new IllegalArgumentException("진행 중인 게임은 삭제할 수 없습니다.");
        }

        if (!room.isSamePassword(password)) {
            throw new IllegalArgumentException("비밀번호가 틀렸습니다.");
        }
        
        roomDao.deleteBy(roomId, password);
    }
```

Reply:
제가 1번을 잘 이해하지 못한 것 같습니다!😅
유효성 검사를 먼저 진행한다는 뜻은, deleteBy를 호출하는 Controller에서 유효성 검사를 한다는 것을 의미하는 걸까요~?

Reply:
요부분이 잘 이해 안된다면 오프라인으로 찾아오세요.